src/protocol/http/src/test/java/org/apache/jmeter/protocol/http/util/TestGraphQLRequestParamUtils.java

```diff
@@ -99,7 +103,15 @@ void testInvalidGraphQLContentType(String contentType) {
 
     @Test
     void testToPostBodyString() throws Exception {
-        assertEquals(EXPECTED_POST_BODY, GraphQLRequestParamUtils.toPostBodyString(params));
+        final JsonNode expectedPostBodyJson = objectMapper.readTree(EXPECTED_POST_BODY);
+        final JsonNode actualPostBodyJson = objectMapper.readTree(GraphQLRequestParamUtils.toPostBodyString(params));
+
+        assertEquals(expectedPostBodyJson.get(GraphQLRequestParamUtils.OPERATION_NAME_FIELD),
+                actualPostBodyJson.get(GraphQLRequestParamUtils.OPERATION_NAME_FIELD));
```

Comment:
Could you please factor this into a assertion method that receives `fieldName`, `expectedPostBody`, and `params`?

Then assertion code would be de-duplicated, and the assertion message could include the source value, so the failure includes not only "expected A got B", but it should include which field failed the comparison, and what was the input data.

Reply:
An alternative option is to use `@ParameterizedTest`, however, please include `field name` and `input data` into the assertion message in any case

Reply:
Hi @vlsi ,
I tried to improve it using `@ParamterizedTest` in 9e63931ee11f206fa846bd8277bfe8b8bd6cb13d. Could you take a look? Thanks!